### PR TITLE
Tob crypto fixes

### PR DESCRIPTION
**Pull request by AdoAdoAdo**

### CODE REVIEW

File: crypto/signing/mcl/multisig/bls.go [hunk 1]
```diff
@@ -281,8 +337,10 @@ func hashPublicKeyPoint(hasher hashing.Hasher, pubKeyPoint crypto.Point) ([]byte
 		return nil, err
 	}
 
-	// H1(pubkey_i)
-	h := hasher.Compute(string(pointBytes))
+	concatPkWithPKs := append(concatPubKeys, pointBytes...)
```
Comment (ccorcoveanu) [inline crypto/signing/mcl/multisig/bls.go]: Shouldn't it be the other way around? pki, {pk1, pk2....}

File: crypto/signing/mcl/multisig/bls.go [hunk 2]
```diff
@@ -133,20 +117,37 @@ func (bms *BlsMultiSigner) VerifyAggregatedSig(
...
+	prepPubKeysPoints := make([]bls.PublicKey, len(pubKeys))
+
+	concatPKs, err := concatPubKeys(pubKeys)
+	if err != nil {
+		return nil, err
+	}
+
+	pubKeysPoints := make([]crypto.Point, len(pubKeys))
+	for i, pubKey := range pubKeys {
+		if check.IfNil(pubKey) {
```
Comment (ccorcoveanu) [inline crypto/signing/mcl/multisig/bls.go]: Why not make this check in the next for and get the point there?

File: crypto/signing/mcl/multisig/bls.go (hunk 1 above)
Comment (AdoAdoAdo) [reply crypto/signing/mcl/multisig/bls.go]: Although it shouldn't matter, indeed the specification is pki, {pk1,...pkn}
done

File: crypto/signing/mcl/multisig/bls.go (hunk 2 above)
Comment (AdoAdoAdo) [reply crypto/signing/mcl/multisig/bls.go]: right, done

**Pull request merged**